.github/actions/slack-check-deployment-blocked/action.yml

```diff
@@ -0,0 +1,57 @@
+name: 'Slack Check Deployment Blocked'
+description: 'Check the Slack #deployments channel title if deployment should be blocked'
+
+inputs:
+  component:
+    description: 'Component being deployed'
+    required: true
+  channel:
+    description: 'Slack channel'
+    required: true
+  slack_token:
+    description: 'Slack bot token'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Get channel info
+      id: slack_channel_info
+      uses: slackapi/slack-github-action@v2.0.0
+      with:
+        method: conversations.info
+        token: ${{ inputs.slack_token }}
+        payload: |
+          channel: ${{ inputs.channel }}
+
+    - name: Get channel topic
+      if: steps.slack_channel_info.outputs.ok == 'true'
+      id: slack_channel_topic
+      shell: bash
+      run: |
+        RESPONSE='${{ steps.slack_channel_info.outputs.response }}'
+        if ! TOPIC=$(echo "$RESPONSE" | jq -r '.channel.topic.value'); then
+          echo "Failed to extract Slack channel topic" >&2
+          exit 0
+        fi
+        echo "topic=$TOPIC" >> $GITHUB_OUTPUT
+
+    - name: Parse channel topic for blocked deployments
+      if: steps.slack_channel_topic.outputs.topic != ''
+      id: parse_channel_topic
+      shell: bash
+      run: |
+        is_blocked() {
+            local raw_component="$1"
+            local component=$(printf '%s' "$raw_component" | sed 's/[][\.*^$/]/\\&/g')
+
+            local status_string="$2"
+            
+            if echo "$status_string" | grep -q ":x:[ ]*${component}[ ]" || echo "$status_string" | grep -q ":x:[ ]*${component}|\$" ; then
```

Comment:
:exploding_head:

Reply:
What is the question ?

Reply:
(grep is limited afaik, | didn't work)

Reply:
No question! Just a lot of magic :)

Reply:
Assumes the human do the right thing as well. Only worry is this interfering with incidents. Is there an easy way to override it?

Reply:
We look for a statement of blocking, otherwise we let it go.
If the slack api fails, we should let it go too (updating the final PR for that).